ts/features/idpay/configuration/store/index.ts

```diff
@@ -0,0 +1,142 @@
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+
+import { createSelector } from "reselect";
+import { getType } from "typesafe-actions";
+import { InstrumentListDTO } from "../../../../../definitions/idpay/InstrumentListDTO";
+import { Action } from "../../../../store/actions/types";
+import { GlobalState } from "../../../../store/reducers/types";
+import { NetworkError } from "../../../../utils/errors";
+import { StatusEnum } from "../../../../../definitions/idpay/InstrumentDTO";
+import {
+  idpayInitiativeInstrumentDelete,
+  idpayInitiativeInstrumentsGet
+} from "./actions";
+
+export type IdPayInitiativeConfigurationState = {
+  instruments: pot.Pot<InstrumentListDTO, NetworkError>;
+  instrumentStatus: Record<string, pot.Pot<StatusEnum, NetworkError>>;
+};
+
+const INITIAL_STATE: IdPayInitiativeConfigurationState = {
+  instruments: pot.none,
+  instrumentStatus: {}
+};
+
+const reducer = (
+  state: IdPayInitiativeConfigurationState = INITIAL_STATE,
+  action: Action
+): IdPayInitiativeConfigurationState => {
+  switch (action.type) {
+    case getType(idpayInitiativeInstrumentsGet.request):
+      return {
+        ...state,
+        instruments: pot.toLoading(pot.none)
+      };
+    case getType(idpayInitiativeInstrumentsGet.success):
+      return {
+        ...state,
+        instruments: pot.some(action.payload)
+      };
+    case getType(idpayInitiativeInstrumentsGet.failure):
+      return {
+        ...state,
+        instruments: pot.toError(state.instruments, action.payload)
+      };
+    case getType(idpayInitiativeInstrumentDelete.request): {
+      const { instrumentId } = action.payload;
+      if (!state.instrumentStatus[instrumentId]) {
+        return {
+          ...state,
+          instrumentStatus: {
+            ...state.instrumentStatus,
+            [instrumentId]: pot.noneLoading
+          }
+        };
+      }
+      return {
+        ...state,
+        instrumentStatus: {
+          ...state.instrumentStatus,
+          [instrumentId]: pot.toLoading(state.instrumentStatus[instrumentId])
+        }
+      };
+    }
+    case getType(idpayInitiativeInstrumentDelete.success): {
+      const { instrumentId } = action.payload;
+      return {
+        instruments: pipe(
+          pot.getOrElse(state.instruments, null),
+          O.fromNullable,
+          O.map(el => el.instrumentList),
+          O.map(instruments =>
+            instruments.filter(
+              instrument => instrument.instrumentId !== instrumentId
+            )
+          ),
+          O.map(filteredInstruments =>
+            pot.some({ instrumentList: filteredInstruments })
+          ),
+          O.getOrElseW(() => pot.none)
+        ),
```

Comment:
I'm not sure about the reason of this part. Why we need to alter the instruments list after the deletion request? After the request the state goes to PENDING_DEACTIVATION, but we still need to have that instrument.

Reply:
I agree with you, in the following commit b0018aed4846f147d1f309915b3f39838e220179 I added a polling handler to maintain updated the status in real time.